syntax = "proto3";

service MusicRecommendationService {
  
  rpc CreateRecommendation (CreateRecommendationRequest) returns (CreateRecommendationResponse);

  rpc GetRecommendation (GetRecommendationRequest) returns (GetRecommendationResponse);

  rpc UpdateRecommendation (UpdateRecommendationRequest) returns (UpdateRecommendationResponse);

  rpc DeleteRecommendation (DeleteRecommendationRequest) returns (DeleteRecommendationResponse);
}

message CreateRecommendationRequest {
  string title = 1;
  string artist = 2;
  string genre = 3;
}

message CreateRecommendationResponse {
  string id = 1;  // Unique ID of the created recommendation
  string message = 2;
}

message GetRecommendationRequest {
  string id = 1;  // ID of the recommendation to retrieve
}

message GetRecommendationResponse {
  string id = 1;
  string title = 2;
  string artist = 3;
  string genre = 4;
}

message UpdateRecommendationRequest {
  string id = 1;
  string title = 2;
  string artist = 3;
  string genre = 4;
}

message UpdateRecommendationResponse {
  string message = 1;
}

message DeleteRecommendationRequest {
  string id = 1;
}

message DeleteRecommendationResponse {
  string message = 1;
}
